examples/machine-learning/README.md

```diff
@@ -0,0 +1,319 @@
+ghpc deploy slurm-a3-cluster --auto-approve
+```
+
+## Receive Data Path Manager (RxDM)
+
+To achieve optimal application performance, an additional service called the
+"Receive Data Path Manager" (RxDM) must run with the same lifetime as the job.
+Additionally, a NCCL plugin must be installed into the execution environment of
+the workload. Both the RxDM and plugin are distributed by Docker container
+images.
+
+This blueprint includes a Slurm "Prolog" and "Epilog" script that will run
+before and after every job running on more than 1 A3 compute node. The Prolog
+will perform the following actions:
+
+- Install the NCCL plugin into /var/lib of the host
+- Run the RxDM service
+  - This is a long-lived service that runs alongside the job
+  - Mounts `/var/lib/nvidia/lib64` into `/usr/lib/nvidia/lib64` of the container
+  - Mount `/opt/tcpdirect_benchmark/` from the host into the container so that a
+  textproto file defining the mapping from GPU to NIC is available. This file
+  is present in the Deep Learning VM (DLVM) images that contain TCPDirect
+  patches.
+  - Mount `/run/tcpx-${SLURM_JOB_ID}` from the container into the host. This is
+  set to the environment variables `${UDS_PATH}` in the script. This directory
+  contains Unix socket files that implement a TCPx interface available to the
+  user workload at `${UDS_PATH}`. The job must be configured to be aware of this
+  path using `NCCL_GPUDIRECTTCPX_UNIX_CLIENT_PREFIX` environment variable!
+
+The Epilog will
+
+- Stop the RxDM service
+- Prune any stopped containers (freeing up disk space)
+- Remove the directory at `${UDS_PATH}`
+
+## Jobs using the RxDM / TCPx
+
+Jobs that are running across multiple A3 VMs will benefit from using the RxDM
+and the NCCL plugin. An example containerized job is located at
+`/opt/apps/scripts/run-nccl-tests.sh`. In addition to setting standard NCCL
+configuration values, a job must:
+
+- Set `NCCL_GPUDIRECTTCPX_UNIX_CLIENT_PREFIX` to `${UDS_PATH}`
+- Set the `LD_LIBRARY_PATH` to include `/var/lib/tcpx/lib64` and `/usr/local/nvidia/lib64`
+
+If job is containerized
+
+- Mount `${UDS_PATH}` into the container at the same path
+- Mount `/var/lib/tcpx/lib64` to `/var/lib/tcpx/lib64` in the container (to make the
+  NCCL plugin available)
+- Paths can be modified if `LD_LIBRARY_PATH` is likewise modified
+
+## Example workload (NCCL benchmark)
+
+The example workload below demonstrates the pattern recommended in Activating
+the Receive Data Path Manager during jobs while running the standard nccl-tests
+benchmark. It assumes the availability of a GPU/NIC topology file at
+`/opt/tcpdirect_benchmark/gpu_rxq_configuration.textproto`. This file is built
+into the DLVM images used by this solution, but may need to be provided if
+using an alternative image.
+
+### Clone the HPC Toolkit repository containing the NCCL benchmark
+
+```shell
+git clone https://github.com/GoogleCloudPlatform/hpc-toolkit
+cd hpc-toolkit/examples/machine-learning/nccl-tests
```

Comment:
Do we have this in repo or we are planning to add in following PRs ?

Reply:
These are contained in #2361